### ISSUE TITLE
Two-way binding `URL <-> $params/$query`?

### ISSUE THREAD

**Issue by Kelin2025** [atomic-router/atomic-router#14]
For now `$query` has only one-way (URL changed -> $query updated) binding
But sometimes we need to sync it back
However, for now updating `$query` store won't trigger URL update (neither from `router` nor from `route`s)

#### Search pages
For example, on some page with dynamic filters/search, we might want to store filters in query
```tsx
searchRoute.$query
  .on(combine({ q: $q, minPrice: $minPrice, maxPrice: $maxPrice }), (prev, next) => next))
```

#### Magic auth links
If you have magic links that authorize you (e.g. "Click here to authorize" email or navigation between 2 subdomains with different frontend), you probably want to remove sensitive data:
```tsx
router.$query.on(router.initialized, ({ token, ...query }) => query)
```

## Controversy 1
For now you can update query only through `someRoute.navigate` with current `$params` and `$query`
But this will trigger a lot of unnecessary computations + semantically it does not seem correct.
In use case №2, you don't even have "a specific" route, you just need to erase token from it, no matter what route is opened

## Controversy 2
Actually I don't like direct updates of `$params` and `$query`. 
I'm sure it should be controlled through `open/navigate` methods, and updates obtained through `opened/updated/closed` events. 
Because listening to `$params` / `$query` will lead to "is this update came from `route.$query` changed by user or it's router sent new query to `route.$query`" problem

Thoughts? 🤔

**Comment by sergeysova** [comment]
Relates #20

**Comment by Kelin2025** [comment]
Released in [**v0.7.1**](https://github.com/atomic-router/atomic-router/releases/tag/v0.7.1) for queries
